tests/components/test_forecast.py

```diff
@@ -46,8 +45,8 @@ def test_setup_bad_api_key(self, mock_get_forecast):
         msg = '400 Client Error: Bad Request for url: {}'.format(url)
         mock_get_forecast.side_effect = HTTPError(msg,)
 
-        with pytest.raises(HTTPError):
-            forecast.setup_platform(self.hass, self.config, MagicMock())
+        response = forecast.setup_platform(self.hass, self.config, MagicMock())
+        self.assertFalse(response)
```

Comment:
Not sure what is convention for hass. I wrapper the portion of the setup that calls forecast.io in a try/catch. In the catch, it logs an error message and just returns false. I've seen some components do that. Others return `None` or `[ ]`

Could just as easily raise an error.